File: app/src/app/model/item.ts
export class Item {
  id: number | null;
  brand: string;
  name: string;
  description: string;

  constructor(item: Partial<Item> = {}) {
    this.id = item?.id || null;
    this.brand = item?.brand || '';
    this.name = item?.name || '';
    this.description = item?.description || '';
  }
}

File: src/main/java/de/toxic2302/inventorymanagementsystem/core/modules/item/entity/Item.java
package de.toxic2302.inventorymanagementsystem.core.modules.item.entity;

import de.toxic2302.inventorymanagementsystem.base.entity.BaseEntity;
import de.toxic2302.inventorymanagementsystem.core.modules.user.entity.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "items")
@Getter @Setter
public class Item extends BaseEntity {

    // ---- Globales ----
    @Column
    @NotNull
    private String name;

    @Column
    @NotNull
    private String brand;

    @Column
    private String serialNumber;

    @Column(length = 10000)
    private String description;

    @ManyToOne
    private User user;

    @Column
    private Boolean sold;

    @Column
    private Integer amount;

    // ---- Constructor ----
    public Item() {}
}
